soc/wch/ch32v00x/soc.c

```diff
@@ -9,46 +9,52 @@
 #include <ch32v003fun.h>
 #include <soc.h>
 
-#define PLL_CSS_HSE_MASK ((uint32_t)0xFEF6FFFF)
-#define HSEBYP_MASK      ((uint32_t)0xFFFBFFFF)
-#define PLLSRC_MASK      ((uint32_t)0xFFFEFFFF)
-
-static void soc_ch32v003_sysclock_hsi(void)
+static int soc_ch32v003_init(void)
 {
-	/* Flash 0 wait state */
-	FLASH->ACTLR = (FLASH->ACTLR & ~FLASH_ACTLR_LATENCY) | FLASH_ACTLR_LATENCY_1;
-
-	/* HCLK = SYSCLK = APB1 */
-	RCC->CFGR0 |= RCC_HPRE_DIV1;
-	/* PLLCLK = HSI * 2 */
-	RCC->CFGR0 = (RCC->CFGR0 & ~RCC_PLLSRC) | RCC_PLLSRC_HSI_Mul2;
-
-	/* Enable PLL */
+#if defined(CONFIG_SOC_WCH_CH32V00X_PLL)
+	/* Disable the PLL before potentially changing the input clocks. */
+	RCC->CTLR &= ~RCC_PLLON;
+#endif
+#if defined(CONFIG_SOC_WCH_CH32V00X_LSI)
+	RCC->RSTSCKR |= RCC_LSION;
+	while ((RCC->RSTSCKR & RCC_LSIRDY) == 0) {
+	}
+#endif
+#if defined(CONFIG_SOC_WCH_CH32V00X_HSI)
+	RCC->CTLR |= RCC_HSION;
+	while ((RCC->CTLR & RCC_HSIRDY) == 0) {
+	}
+#endif
+#if defined(CONFIG_SOC_WCH_CH32V00X_HSE)
+	RCC->CTLR |= RCC_HSEON;
+	while ((RCC->CTLR & RCC_HSERDY) == 0) {
+	}
+#endif
+#if defined(CONFIG_SOC_WCH_CH32V00X_HSE_AS_PLLSRC)
+	RCC->CFGR0 &= ~RCC_PLLSRC;
+#elif defined(CONFIG_SOC_WCH_CH32V00X_HSI_AS_PLLSRC)
+	RCC->CFGR0 &= ~RCC_PLLSRC;
+#endif
+#if defined(CONFIG_SOC_WCH_CH32V00X_PLL)
 	RCC->CTLR |= RCC_PLLON;
 	while ((RCC->CTLR & RCC_PLLRDY) == 0) {
 	}
-
-	/* Select PLL as system clock source */
+#endif
+#if defined(CONFIG_SOC_WCH_CH32V00X_HSI_AS_SYSCLK)
+	RCC->CFGR0 = (RCC->CFGR0 & ~RCC_SW) | RCC_SW_HSI;
+#elif defined(CONFIG_SOC_WCH_CH32V00X_HSE_AS_SYSCLK)
+	RCC->CFGR0 = (RCC->CFGR0 & ~RCC_SW) | RCC_SW_HSE;
+#elif defined(CONFIG_SOC_WCH_CH32V00X_PLL_AS_SYSCLK)
 	RCC->CFGR0 = (RCC->CFGR0 & ~RCC_SW) | RCC_SW_PLL;
-	while ((RCC->CFGR0 & RCC_SWS) != RCC_SWS_PLL) {
-	}
-}
+#endif
+	RCC->CTLR |= RCC_CSSON;
 
-static int soc_ch32v003_init(void)
-{
-	RCC->CTLR |= RCC_HSION;
-	/* PLLSRC = 0, which is the HSI */
-	RCC->CFGR0 = RCC_MCO_NOCLOCK & ~RCC_PLLSRC;
-	/* Turn off PLLON, CSSON, HSEON */
-	RCC->CTLR &= PLL_CSS_HSE_MASK;
-	/* Turn off HSEBYP */
-	RCC->CTLR &= HSEBYP_MASK;
-	/* Turn off PLLSRC again? */
-	RCC->CFGR0 &= PLLSRC_MASK;
-	/* Write to INTR to clear the CSSC, PLLRDYC, HSERDYC, and LSIRDYC flags */
-	RCC->INTR = 0x009F0000;
-
-	soc_ch32v003_sysclock_hsi();
+	/* Clear the interrupt flags. */
+	RCC->INTR = RCC_CSSC | RCC_PLLRDYC | RCC_HSERDYC | RCC_LSIRDYC;
+	/* Set the Flash to 0 wait state */
+	FLASH->ACTLR = (FLASH->ACTLR & ~FLASH_ACTLR_LATENCY) | FLASH_ACTLR_LATENCY_1;
+	/* HCLK = SYSCLK = APB1 */
+	RCC->CFGR0 = (RCC->CFGR0 & ~RCC_HPRE) | RCC_HPRE_DIV1;
```

Comment:
not ideal to introduce a file just to change it later in the same patch series, could you rework the commit so that this is introduced in the "final" version directly? Just empty out the stubs where they are introduced I'd say, or fold  them back, whatever works just avoid changing them half series, this is confusing for the reviewers.

Reply:
I was just thinking about this earlier today! This PR can't have actuals `+-` diffs in it. I will fix this now. Thanks!

Reply:
Fixed now - thanks!